websites/R/Robloxforum/presence.ts

```diff
@@ -0,0 +1,77 @@
+const presence = new Presence({
+		clientId: "993929577099448481",
+	}),
+	date = Date.now();
+
+presence.on("UpdateData", async () => {
+	if (document.location.pathname === "/") {
+		const presenceData: PresenceData = {
+			details: "Viewing the homepage",
+			smallImageKey: "small",
+			smallImageText: "Swoosh",
+			largeImageKey: "ig-mal",
+			startTimestamp: date,
+		};
+		presence.setActivity(presenceData);
+	} else if (document.location.pathname.startsWith("/members")) {
+		const presenceData: PresenceData = {
+			details: "Looking at a profile",
+			state: `Profile: ${document
+				.querySelector("meta[property='og:title']")
+				.getAttribute("content")}`,
+			smallImageKey: "small",
+			smallImageText: "Profiles?!",
+			largeImageKey: "ig-mal",
+			startTimestamp: date,
+		};
+		presence.setActivity(presenceData);
```

Comment:
don't do this. Make a presenceData object at the top and edit it as you go and only set it at the bottom of the listener

Reply:
I'm not sure on what you mean -- how would I do this?

Reply:
I literally told you what you needed to do, I don’t know how to be more specific than that

Reply:
I'm not sure how you would do this coding wise? I always get the error "presenData was declared but not read" or something along this lines (currently don't got the error open)